src/bin/cargo.rs

```diff
@@ -309,7 +311,7 @@ fn execute_external_subcommand(config: &Config, cmd: &str, args: &[String]) -> C
     let command = match path {
         Some(command) => command,
         None => {
-            return Err(human(match find_closest(config, cmd) {
+            return Err(CargoError::from(match find_closest(config, cmd) {
```

Comment:
Ah unfortunately `into` doesn't work w/ turbofish as the type parameter is in the wrong place, if `.into()` doesn't work then `CargoError::from` is fine, no worries!